### PR TITLE
bump: joonix/log to NewFormater()

### LINKED ISSUE #3341: Changed API of joonix/log
**What happened**:

when running `go build` with a project that uses agones 1.34.0 with a newer verison of `joonix/log`. It uses `joonix.FluentdFormatter` in `pkg/util/runtime/runtime` which changed now to `joonix.NewFormatter()` resulting in the following compilation error:

```bash
# agones.dev/agones/pkg/util/runtime
../../go/pkg/mod/agones.dev/agones@v1.34.0/pkg/util/runtime/runtime.go:39:30: undefined: joonix.FluentdFormatter
```

**What you expected to happen**:

building successfully. 

**How to reproduce it (as minimally and precisely as possible)**:

use i think any agenes project with the go agones and run `go get -u ./...` or just update the joonix/log package. 

**Anything else we need to know?**:

Of course building works when using the specified version of joonix/log in the go.mod `go mod edit -require github.com/joonix/log@v0.0.0-20180502111528-d2d3f2f4a806`

**Environment**:
- Agones version: 1.34.0
- Kubernetes version (use `kubectl version`): none
- Cloud provider or hardware configuration: none
- Install method (yaml/helm): none/golang `go get agones.dev/agones`
- Troubleshooting guide log(s):
- Others:

### PR DESCRIPTION
**Pull request by JohnnyS318**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> 
/kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

Bump the version of joonix/log to the (new) `joonix.NewFormatter()`.

I would suggest to remove the dependency all together. But at least keep the version up to date.

**Which issue(s) this PR fixes**:

Closes #3341

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/util/runtime/runtime.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ type stackTracer interface {
 
 // replace the standard glog error logger, with a logrus one
 func init() {
-	logrus.SetFormatter(&joonix.FluentdFormatter{})
+	logrus.SetFormatter(joonix.NewFormatter())
```
Comment (aimuz) [inline pkg/util/runtime/runtime.go:39]: Do the fields match exactly?

Comment (JohnnyS318) [reply pkg/util/runtime/runtime.go]: the fluentd format only uses three field: "time", "severity", "message". The issue [https://github.com/joonix/log/issues/2](https://github.com/joonix/log/issues/2) sets the default JSONFormatter with a FieldMap as a alternative.

**Comment by aimuz** [comment]
Can you give the log output before this change, as well as the log output after the change, for review?

**Comment by JohnnyS318** [comment]
Actually then most of the dependency upgrades is unnecessary because I removed the dependency to joonix/log.

**Comment by aimuz** [comment]
If the log output is not affected, I believe this is a good change.

**Comment by JohnnyS318** [comment]
> Can you give the log output before this change, as well as the log output after the change, for review?

Yeah let me first try to run it where i can see some logs.

**Comment by aimuz** [comment]
unchanged

```
{"message":"Player Disconnected","playerID":"1234","sdkName":"go","severity":"info","source":"*sdkserver.LocalSDKServer","time":"2023-08-23T00:27:37.928895153Z"}
{"message":"Gameserver update received","sdkName":"go","severity":"info","source":"*sdkserver.LocalSDKServer","time":"2023-08-23T00:27:37.928930197Z"}
```

changed

```
{"message":"Player Disconnected","playerID":"1234","sdkName":"go","severity":"info","source":"*sdkserver.LocalSDKServer","time":"2023-08-23T07:08:31Z"}
{"message":"Gameserver update received","sdkName":"go","severity":"info","source":"*sdkserver.LocalSDKServer","time":"2023-08-23T07:08:31Z"}
```

Looks like the format of the log has changed, Can you make it consistent?

@JohnnyS318

File: pkg/util/runtime/runtime.go [hunk 2]
```diff
@@ -36,7 +35,13 @@ type stackTracer interface {
 
 // replace the standard glog error logger, with a logrus one
 func init() {
-	logrus.SetFormatter(&joonix.FluentdFormatter{})
+	logrus.SetFormatter(&logrus.JSONFormatter{
```
Comment (aimuz) [inline pkg/util/runtime/runtime.go:38]: We should keep the time format the same as before

**Review by aimuz — CHANGES_REQUESTED** [review]

**Comment by JohnnyS318** [comment]
> unchanged
> 
> ```
> {"message":"Player Disconnected","playerID":"1234","sdkName":"go","severity":"info","source":"*sdkserver.LocalSDKServer","time":"2023-08-23T00:27:37.928895153Z"}
> {"message":"Gameserver update received","sdkName":"go","severity":"info","source":"*sdkserver.LocalSDKServer","time":"2023-08-23T00:27:37.928930197Z"}
> ```
> 
> changed
> 
> ```
> {"message":"Player Disconnected","playerID":"1234","sdkName":"go","severity":"info","source":"*sdkserver.LocalSDKServer","time":"2023-08-23T07:08:31Z"}
> {"message":"Gameserver update received","sdkName":"go","severity":"info","source":"*sdkserver.LocalSDKServer","time":"2023-08-23T07:08:31Z"}
> ```
> 
> Looks like the format of the log has changed, Can you make it consistent?
> 
> @JohnnyS318

Should now be fixed. The FluentDFormatter used the RFC3339Nano timestamp format

**Review by aimuz — APPROVED** [review]
/lgtm 

@markmandel I think this is a good change, do you have time to look at it?

**Comment by markmandel** [comment]
LGTM, just dragging @gongmax and @zmerlynn in to triple check, since logger changes have such a wide impact.

I don't think the slight change in time format should matter.

`"time":"2023-08-23T00:27:37.928930197Z"` vs ``"time":"2023-08-23T07:08:31Z"` - Not sure we are required to go to the sub-nano second here 😃

**Review by gongmax — APPROVED** [review]
LGTM

**Comment by aimuz** [comment]
Is it possible to add a `retest` command? For quickly retrying incorrect tests

**Pull request merged**